pkg/client/client_v2.go

```diff
@@ -781,18 +780,29 @@ func (c *clientV2) tryPerformDiagnostics(actionResults chan *proto.ActionRespons
 }
 
 func (c *clientV2) registerDefaultDiagnostics() {
-	c.registerPprofDiagnostics("goroutine", "stack traces of all current goroutines")
-	c.registerPprofDiagnostics("heap", "a sampling of memory allocations of live objects")
-	c.registerPprofDiagnostics("allocs", "a sampling of all past memory allocations")
-	c.registerPprofDiagnostics("threadcreate", "stack traces that led to the creation of new OS threads")
-	c.registerPprofDiagnostics("block", "stack traces that led to blocking on synchronization primitives")
-	c.registerPprofDiagnostics("mutex", "stack traces of holders of contended mutexes")
+	c.registerPprofDiagnostics("goroutine", "stack traces of all current goroutines", 0)
+	c.registerPprofDiagnostics("heap", "a sampling of memory allocations of live objects", 0)
+	c.registerPprofDiagnostics("allocs", "a sampling of all past memory allocations", 0)
+	c.registerPprofDiagnostics("threadcreate", "stack traces that led to the creation of new OS threads", 0)
+	c.registerPprofDiagnostics("block", "stack traces that led to blocking on synchronization primitives", 0)
+	c.registerPprofDiagnostics("mutex", "stack traces of holders of contended mutexes", 0)
 }
 
-func (c *clientV2) registerPprofDiagnostics(name string, description string) {
-	c.RegisterDiagnosticHook(name, description, fmt.Sprintf("%s.txt", name), "plain/text", func() []byte {
+func (c *clientV2) registerPprofDiagnostics(name string, description string, debug int) {
```

Comment:
The agent depends on elastic-agent-client, maybe we could keep this implementation here so it isn't duplicated from https://github.com/elastic/elastic-agent-client/pull/69/?